### PR TITLE
test: retrieve apm-server log during smoke tests and move config to template

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

Improve the current smoke test for supported os:
- switch to terraform `templatefile` and move the template to a separate file
- retrieve apm-server logs with a destroy time provisioner

Log retrieval is currently broken because of https://github.com/elastic/apm-server/issues/10619

This PR is the result of backporting the work on https://github.com/elastic/apm-server/issues/10467 to the current supported-os smoke test

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

1. `EC_API_KEY=<api-key> TEST_DIR=testing/smoke/supported-os/ make smoketest/run`
1. smoke tests are passing

### CODE REVIEW

File: testing/smoke/supported-os/main.tf [hunk 1]
```diff
@@ -164,6 +175,19 @@ resource "aws_instance" "apm" {
...
 
+resource "null_resource" "getapmserverlogs" {
+  triggers = {
+    user        = local.image_ssh_users[var.aws_os]
+    host        = aws_instance.apm.public_ip
+    private_key = aws_key_pair.provisioner_key.key_name
+  }
+
+  provisioner "local-exec" {
+    when    = destroy
```
Comment (marclop) [inline testing/smoke/supported-os/main.tf:186]: Nice!

**Review by marclop — COMMENTED** [review]
Looks good, should we also update the CI pipeline to store that log file as an artifact?

**Comment by kruskall** [comment]
> should we also update the CI pipeline to store that log file as an artifact?

That's a great idea!

smoke tests are being moved to github actions so I think we should do it in https://github.com/elastic/apm-server/pull/10654. WDYT ?

File: testing/smoke/supported-os/main.tf [hunk 2]
```diff
@@ -164,6 +175,19 @@ resource "aws_instance" "apm" {
   vpc_security_group_ids = [aws_security_group.main.id]
 }
 
+resource "null_resource" "getapmserverlogs" {
```
Comment (marclop) [inline testing/smoke/supported-os/main.tf:178]: I haven't tested this, will it always run before the `aws_instance` is destroyed?

Comment (marclop) [reply testing/smoke/supported-os/main.tf:178]:
```suggestion
resource "null_resource" "apm_server_log" {
```

**Review by marclop — APPROVED** [review]
@kruskall could you create an issue about doing that or make sure that gets done, so we don't forget about it?

**Pull request merged**